cli/lib/kontena/cli/cloud/login_command.rb

```diff
@@ -137,12 +137,11 @@ def web_flow
       puts
 
       server_thread  = Thread.new { Thread.main['response'] = web_server.serve_one }
-      browser_thread = Thread.new { Launchy.open(uri.to_s) }
+      Kontena::Cli::BrowserLauncher.open(uri.to_s)
```

Comment:
This is assuming that the browser launch command just opens up a new tab/window in the background and exits. That's probably the case 99% of the time, but testing with the browser not already running shows that at least `sensible-browser` will actually end up spawning a new browser process and block waiting for it to exit... that might count as a bug in `sensible-browser`, though.

Either way, yes, it's better for this to explode immediately on errors, rather than hanging in the `spinner` -> `server_thread.join` that will never happen if the browser fails to launch.